src/ApiKitClient.ts

```diff
@@ -0,0 +1,56 @@
+import axios, { AxiosError, AxiosRequestConfig } from 'axios';
+import { HttpMethod, ApiClientResponse, PaginatedResponse } from './types';
+
+export function ApiKitClient<T>(
+  baseUrl: string,
+  authToken?: string,
+  handleUnauthorizedAccess?: () => void
+) {
```

Comment:
authToken optional ?

Reply:
I think I missed this!
We have to agree on if the unauthorization callback will be optional as well or not!